src/transformers/modeling_flax_utils.py

```diff
@@ -820,31 +891,35 @@ def from_pretrained(
                 _commit_hash=commit_hash,
             )
 
+        safetensors_from_pt = False
+        if filename == SAFE_WEIGHTS_NAME:
+            with safe_open(resolved_archive_file, framework="flax") as f:
+                safetensors_metadata = f.metadata()
+            if safetensors_metadata is None or safetensors_metadata.get("format") not in ["pt", "tf", "flax"]:
+                raise OSError(
+                    f"The safetensors archive passed at {resolved_archive_file} does not contain the valid metadata."
+                    " Make sure you save your model with the `save_pretrained` method."
+                )
+            safetensors_from_pt = safetensors_metadata.get("format") == "pt"
+            safetensors_metadata.get("format") == "tf"
+
         # init random models
         model = cls(config, *model_args, _do_init=_do_init, **model_kwargs)
 
-        if from_pt:
+        if safetensors_from_pt:
+            state_dict = {}
+            with safe_open(resolved_archive_file, framework="pt") as f:
```

Comment:
(nit) Should this loading of the torch tensors go into the function `load_pytorch_state_dict_in_flax_state_dict`? This would be consistent with `load_pytorch_checkpoint_in_flax_state_dict` where we load the PT state dict inside the function: https://github.com/huggingface/transformers/blob/84724efd101af52ed3d6af878e41ff8fd651a9cc/src/transformers/modeling_flax_pytorch_utils.py#L59